### PR TITLE
Get range bounds of range actions

### PR DESCRIPTION
**Pull request by alexandrem-rte**
**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [ ] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem ?** *If so, link to this issue using `'#XXX'` and skip the rest*

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*

**What is the current behavior?** *(You can also link to an open issue here)*

**What is the new behavior (if this is a feature change)?**

**Does this PR introduce a breaking change?** *(What changes might users need to make in their application due to this PR?)*

**Other information**:

(if any of the questions/checkboxes don't apply, please delete them entirely)

### CODE REVIEW

File: data/crac-file/crac-impl/src/main/java/com/farao_community/farao/data/crac_impl/remedial_action/range_action/PstRange.java [hunk 1]
```diff
@@ -79,25 +105,23 @@ public double getMaxValueWithRange(Network network, Range range) {
...
+                return initialTapPosition + extremumValue;
             case RELATIVE_DYNAMIC:
-                throw new FaraoException("RelativeDynamicRanges are not handled for the moment");
-        }
-        switch (range.getRangeDefinition()) {
-            case STARTS_AT_ONE:
-                return lowTapPosition + extremumValueWithRange;
-            case CENTERED_ON_ZERO:
-                return extremumValueWithRange;
+                // return getCurrentTapPosition(network) + extremumValue;
```
Comment (terriervik) [inline data/crac-file/crac-impl/src/main/java/com/farao_community/farao/data/crac_impl/remedial_action/range_action/PstRange.java]: Please delete this line

File: data/crac-file/crac-impl/src/main/java/com/farao_community/farao/data/crac_impl/remedial_action/range_action/PstRange.java [hunk 2]
```diff
@@ -79,25 +105,23 @@ public double getMaxValueWithRange(Network network, Range range) {
     }
 
     private double getExtremumValueWithRange(Network network, Range range, double extremumValue) {
```
Comment (terriervik) [inline data/crac-file/crac-impl/src/main/java/com/farao_community/farao/data/crac_impl/remedial_action/range_action/PstRange.java]: network has become useless

File: data/crac-file/crac-impl/src/test/java/com/farao_community/farao/data/crac_impl/remedial_action/range_action/PstRangeTest.java [hunk 3]
```diff
@@ -63,19 +85,40 @@ public void setUp() throws Exception {
         Mockito.when(range2.getRangeDefinition()).thenReturn(RangeDefinition.STARTS_AT_ONE);
         Mockito.when(range2.getMin()).thenReturn(7.);
 
-        NetworkElement networkElement = Mockito.mock(NetworkElement.class);
+        networkElement = Mockito.mock(NetworkElement.class);
+        Mockito.when(networkElement.getId()).thenReturn(networkElementId);
 
+        // clarifier l'usage qu'on a de pstRange et pstRange1
```
Comment (terriervik) [inline data/crac-file/crac-impl/src/test/java/com/farao_community/farao/data/crac_impl/remedial_action/range_action/PstRangeTest.java]: :gb: :us: :canada: :australia: => :ok:
:fr: :de: :sweden: :es: => :no_entry_sign:

File: data/crac-file/crac-impl/src/test/java/com/farao_community/farao/data/crac_impl/remedial_action/range_action/PstRangeTest.java [hunk 4]
```diff
@@ -146,19 +189,69 @@ public void applyOnTransformerWithNoPhaseShifter() {
...
+        assertEquals(pstLowTapPosition + absoluteStartOneRangeMin - 1, pstRangeWithAbsoluteRange.getMinValue(network), 0);
+    }
+
+    @Test
+    public void pstWithAbsoluteCenteredZeroRange() {
+        PstRange pstRangeWithAbsoluteRange = new PstRange("id", networkElement);
+        pstRangeWithAbsoluteRange.addRange(absoluteCenteredZeroRange);
+        pstRangeWithAbsoluteRange.setReferenceValue(network);
+        assertEquals((pstLowTapPosition + pstHighTapPosition) / 2 + absoluteCenteredZeroRangeMax, pstRangeWithAbsoluteRange.getMaxValue(network), 0);
+        assertEquals((pstLowTapPosition + pstHighTapPosition) / 2 + absoluteCenteredZeroRangeMin, pstRangeWithAbsoluteRange.getMinValue(network), 0);
```
Comment (terriervik) [inline data/crac-file/crac-impl/src/test/java/com/farao_community/farao/data/crac_impl/remedial_action/range_action/PstRangeTest.java]: integer division in floating-point context

File: data/crac-file/crac-impl/src/main/java/com/farao_community/farao/data/crac_impl/remedial_action/range_action/PstRange.java [hunk 5]
```diff
@@ -130,10 +154,4 @@ private PhaseTapChanger checkValidPstAndGetPhaseTapChanger(Network network) {
         }
         return phaseTapChanger;
     }
-
-    public int getCurrentTapPosition(Network network) {
-        TwoWindingsTransformer transformer = network.getTwoWindingsTransformer(networkElement.getId());
-        PhaseTapChanger phaseTapChanger = transformer.getPhaseTapChanger();
-        return phaseTapChanger.getTapPosition();
```
Comment (terriervik) [inline data/crac-file/crac-impl/src/main/java/com/farao_community/farao/data/crac_impl/remedial_action/range_action/PstRange.java]: Wasn't it a nice method? :cry:

File: data/crac-file/crac-impl/src/main/java/com/farao_community/farao/data/crac_impl/remedial_action/range_action/PstRange.java [hunk 6]
```diff
@@ -52,23 +60,41 @@ public PstRange(String id, String name, String operator, NetworkElement networkE
...
     @Override
     public void desynchronize() {
-        lowTapPosition = (int) Double.NaN;
+        currentTapPosition = (int) Double.NaN;
+    }
+
+    public void setReferenceValue(Network network) {
+        synchronize(network);
+        initialTapPosition = currentTapPosition;
+        TwoWindingsTransformer transformer = network.getTwoWindingsTransformer(networkElement.getId());
```
Comment (terriervik) [inline data/crac-file/crac-impl/src/main/java/com/farao_community/farao/data/crac_impl/remedial_action/range_action/PstRange.java]: Do we need to define this object?

File: data/crac-file/crac-impl/src/test/java/com/farao_community/farao/data/crac_impl/remedial_action/range_action/PstRangeTest.java (hunk 3 above)
Comment (alexandrem-rte) [reply data/crac-file/crac-impl/src/test/java/com/farao_community/farao/data/crac_impl/remedial_action/range_action/PstRangeTest.java]: I deleted many useless statements in the setup method

File: data/crac-file/crac-impl/src/main/java/com/farao_community/farao/data/crac_impl/remedial_action/range_action/PstRange.java (hunk 5 above)
Comment (alexandrem-rte) [reply data/crac-file/crac-impl/src/main/java/com/farao_community/farao/data/crac_impl/remedial_action/range_action/PstRange.java]: I think it is better to access the network once in the synchronize method, and to save the value of the current tap position in an attribute. Moreover it makes it possible to remove an argument from some methods signatures.

File: data/crac-file/crac-impl/src/main/java/com/farao_community/farao/data/crac_impl/remedial_action/range_action/PstRange.java (hunk 6 above)
Comment (alexandrem-rte) [reply data/crac-file/crac-impl/src/main/java/com/farao_community/farao/data/crac_impl/remedial_action/range_action/PstRange.java]: no, I removed it

**Pull request merged**